os/drivers/wireless/realtek/os/tizenrt/tizenrt_service.c

```diff
@@ -43,9 +44,28 @@ void timer_wrapper(_timerHandle timer_hdl)
 		timer_entry->timeout = 0;
 		DBG_TRACE("[%d]: TTTTimeout timer=%p\n", rtw_get_current_time(), timer_entry->timer);
 #endif	
-		//ndbg("\r\n==============>>timer->function = %x\r\n",timer_entry->timer->function);
-		if(timer_entry->timer->function)
+		if(timer_entry->timer->data){
+			struct _ADAPTER *_adapter = (struct _ADAPTER *)(timer_entry->timer->data);
+			if(_adapter->RxStop){
+				DBG_871X("[timer_wrapper]:RxStop = %d\n",_adapter->RxStop);
+				return;
+			}
+		}else{
+			DBG_871X("[timer_wrapper]:Adapter = NULL\n");
+			return;
+		}
+		
+		timer_entry->timer->timevalue -= 1000;
+		
+		if(timer_entry->timer->timevalue > 0){
+			if(rtw_timerChangePeriod(timer_entry->timer, rtw_ms_to_systime(1000), TIMER_MAX_DELAY) == _FAIL)
+				DBG_ERR("Fail to set timer period");
+		}else if(timer_entry->timer->function){

```

Comment:
same as above.
and after if also 
like: } else if (timer_entry->timer->function) {